code/turf/walls.dm

```diff
@@ -340,6 +340,32 @@ TYPEINFO(/turf/simulated/wall)
 	logTheThing(LOG_STATION, user, "deconstructed a wall ([src.name]) using \a [W] at [get_area(user)] ([log_loc(user)])")
 	dismantle_wall()
 
+/turf/simulated/wall/bullet_act(obj/projectile/P)
+	..()
+	if (!istype(P.proj_data, /datum/projectile/bullet))
+		return
+	var/datum/projectile/bullet/bullet = P.proj_data
+	if (!bullet.ricochets)
+		return
+	if (prob(90))
+		return
+
+	var/proj_name = P.name
+	var/obj/projectile/p_copy = SEMI_DEEP_COPY(P)
```

Comment:
this feels cursed. Is there really not a better way?

Reply:
Not sure really. I was trying it without the copy, modifying the variables on the reflected projectile, and it was compounding for bullets in the rest of the magazine - first shot 1/2 the distance, second shot 1/4 distance, and so on